federationapi/storage/shared/storage_edus.go

```diff
@@ -20,10 +20,19 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"time"
 
 	"github.com/matrix-org/gomatrixserverlib"
 )
 
+// expireEDUTypes contains EDUs which can/should be expired after a given time
+// if the target server isn't reachable for some reason.
+var expireEDUTypes = map[string]struct{}{
```

Comment:
Suggest we might want to make this `map[string]time.Duration` instead, so that we can specify different expiry times for different EDU types?

For example, a read receipt is useful to keep around for longer, but a typing notification isn't really useful if it doesn't get sent within under a minute, or a presence update in under 5/10 minutes.